### PR TITLE
added ShuffleNet model

### PR DESCRIPTION
**Pull request by jaxony**
@Cysu The ShuffleNet implementation learns on MNIST and gets ~94% accuracy after a couple epochs: not great performance (didn't do hyperparam optimisation) but at least it shows the implementation is working.

Would be awesome if you could see if you can get ShuffleNet working with Re-ID as you have more experience than me in this area! Thanks :)

### CODE REVIEW

**Comment by Cysu** [comment]
@jaxony Thank you very much for the ShuffleNet! The code looks good to me. I will checkout and tune the training parameters for it on re-id datasets. It might take some days as there are actually no published results to compare with.

**Comment by jaxony** [comment]
@Cysu Just wondering if you've had any luck in training the ShuffleNet?

**Comment by Cysu** [comment]
@jaxony I apologize for my absence these days. I just came back from CVPR 17'. Will work on this week. Thank you very much again for the patience.
